content/posts/2025-08-31-getting-offer-for-ml-engineer-at-spotify.md

```diff
@@ -0,0 +1,70 @@
+
+The interview process for me in short was:
+1. Call with HR
+2. A 1-hour "Machine Learning Tech Screen"
+3. Assessment day with five interviews:
+   - ML Data Interview 
+   - Values
+   - ML System Design 
+   - ML Depth Interview
+
+In the end I did an extra ML depth interview, but the normal process is probably the ones above.
+
+For me, I broke the preparation down into two parts: the tech screen and assessment day, and did different preparations for each of them. For both of them I tried to Google and YouTube as much as possible to get inspiration for how the interview could look like. This was not to find the exact questions or a script of what to say, but rather more to get a feel for the interview and start thinking about how I would handle the questions. There are A LOT of videos to find out there; however, in hindsight, none of them really matched what I experienced at Spotify. In this post I will share how I prepared and my experience. I will not share any of the interview questions.
+
+## The Tech Screen Preparations
+
+For the tech screen part, I felt that my biggest gap was the coding problem I knew would be in the interview. I got information that the interview would be three parts: a short intro, technical ML questions, and finally a coding part. I felt confident in the first two parts so focused all my energy on the coding part.
+
+There are a ton of different pages to use to train. I used two different ones - feel free to pick whatever you want, I believe there are a ton of good ones out there:
+1. [Leetcode.com](https://leetcode.com/)
+2. [AlgoExpert](https://www.algoexpert.io/product)
+
+What was important for me was to be able to train on the same type of problems/algorithms in different ways. This was to learn the pattern rather than learn the exact questions. I also decided to focus on easy to medium ones; I never did any hard ones, to be honest.
+
+For LeetCode I used the free path, but for AlgoExpert I paid, which fit me really well. I also tried to combine deep understanding with doing bulk problems - doing 1-3 a day for medium ones.
+
+However, the most important part for me was asking friends for help. Solving the problem is one part of the problem, but doing it in front of another person and explaining at the same time is something totally different. I can highly recommend this - also ask for feedback from your friends. If you don't have any programming friends, that's fine. When you solve the problem, you should be able to explain it in a way so anyone can understand. For me, there was a really steep learning curve here and I felt a massive improvement after doing 5-ish.
```

Comment:
"asking friends" => potentially ask friends to do mock interviews with me.

Reply:
Maybe any friend is enough since the explanations should be simple and more focus on logic rather than code.